From ee836ea0c84b89cc28450c6098b7feada770b10c Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 30 Jul 2014 15:40:58 -0400
Subject: [kernel] define u8, s8, u32, etc. limits

Message-id: <1406734861-25152-5-git-send-email-rclark@redhat.com>
Patchwork-id: 86839
O-Subject: [RHEL7 drm 4/7] kernel.h: define u8, s8, u32, etc. limits
Bugzilla: 1119339
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

RHBZ: 1119339
Backport from upstream

commit 89a0714106aac7309c7dfa0f004b39e1e89d2942
Author:     Alex Elder <alex.elder@linaro.org>
AuthorDate: Thu Jan 23 15:54:00 2014 -0800
Commit:     Linus Torvalds <torvalds@linux-foundation.org>
CommitDate: Thu Jan 23 16:36:54 2014 -0800

    kernel.h: define u8, s8, u32, etc. limits

    Create constants that define the maximum and minimum values
    representable by the kernel types u8, s8, u16, s16, and so on.

    Signed-off-by: Alex Elder <elder@linaro.org>
    Cc: Sage Weil <sage@inktank.com>
    Cc: David Miller <davem@davemloft.net>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

CC: Michael S. Tsirkin <mst@redhat.com>
CC: Vlad Yasevich <vyasevic@redhat.com>
CC: Jason Wang <jasowang@redhat.com>
CC: Prarit Bhargava <prarit@redhat.com>
Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/kernel.h b/include/linux/kernel.h
index cca262c..5043fc1 100644
--- a/include/linux/kernel.h
+++ b/include/linux/kernel.h
@@ -29,6 +29,19 @@
 #define ULLONG_MAX (~0ULL)
 #define SIZE_MAX (~(size_t)0)
 
+#define U8_MAX  ((u8)~0U)
+#define S8_MAX  ((s8)(U8_MAX>>1))
+#define S8_MIN  ((s8)(-S8_MAX - 1))
+#define U16_MAX  ((u16)~0U)
+#define S16_MAX  ((s16)(U16_MAX>>1))
+#define S16_MIN  ((s16)(-S16_MAX - 1))
+#define U32_MAX  ((u32)~0U)
+#define S32_MAX  ((s32)(U32_MAX>>1))
+#define S32_MIN  ((s32)(-S32_MAX - 1))
+#define U64_MAX  ((u64)~0ULL)
+#define S64_MAX  ((s64)(U64_MAX>>1))
+#define S64_MIN  ((s64)(-S64_MAX - 1))
+
 #define STACK_MAGIC 0xdeadbeef
 
 #define REPEAT_BYTE(x) ((~0ul / 0xff) * (x))
-- 
1.7.1